HUGCHAT Tool

Aminolroaya opened the pull request:
I added unit test for hugchat tool, integration test can also be added. Hugchat will not work currently without cookies.

20001LastOrder commented:
Thanks for the PR. could you provide a bit more information (maybe a link) about how I can find the environment variables for hugging face and test it locally?

20001LastOrder commented:
Nevermind, I realized it's just the password and email...

20001LastOrder commented on src/tests/unit_tests/tools/test_hug_chat_tool.py:
```diff
@@ -0,0 +1,6 @@
+from sherpa_ai.tools import HugChatTool
```
Please add a new line at the end and add `@external_api` before the test

20001LastOrder commented on src/sherpa_ai/tools.py:
```diff
@@ -209,3 +218,31 @@ def _run(self, query: str) -> str:
+        cookies = sign.login()
+
+        # Save cookies to the local directory
+        cookie_path_dir = "./cookies_snapshot"
```
I think even if we cannot avoid the cookies, we don't need to save them locally. We can remove line 236 and 237

20001LastOrder reviewed (changes requested):
Thanks for the PR. Just some minor comments, please see them below.

oshoma commented:
Hi @Aminolroaya @20001LastOrder  I made some changes as follows:

- enable offline testing with omcks
- enable online testing 
- reformat 
- get Poetry in synch with `main` 

Could you both please try running the following tests to be sure everything works and report back here? If you have Hugging Chat credentials in your .env file then you should get 11 passing tests. Without credentials you should get 10 passing tests and 1 skipped test.

```
poetry install
pytest tests/unit_tests/tools/test_hug_chat_tool.py   -v
```

And if that works, try `pytest tests` too. Here's my output: `1 failed, 56 passed, 2 skipped, 13 warnings`. 
The failure is expected, it will be resolved when we fix #252.

@20001LastOrder If tests run clean and your code review is OK then I think with these changes we are ready to merge into `main`. We should do a squash merge.

Aminolroaya commented:
@oshoma apologies for the late reply. Is there anything that needs to be taken care of before merging? cc @amirfz

amirfz commented:
@Aminolroaya I think there are some unaddressed comments from Osh on this PR. have you seen those?

Aminolroaya commented:
> @Aminolroaya I think there are some unaddressed comments from Osh on this PR. have you seen those?

Yes. Just want to check if using the Hugchat tool is still considered an enhancement for the project. If yes, I will look into it this weekend. tnx.

20001LastOrder commented:
@Aminolroaya I think this is definitely still valuable. Especially now we are planning on fully configurable agents, adding a new tool to an agent will be much easier. Thus, an additional tool would be nice for different use cases.

20001LastOrder commented:
Closed as not planned
